pkg/collector/pipeline.go

```diff
@@ -69,23 +93,24 @@ func (p *Pipeline) AddProcessor(processor ReportProcessor) {
 
 // ProcessReports extracts reports from a POST upload payload, as defined by the
 // Reporting spec, and runs all of the processors in the pipeline against each
-// report.
-func (p *Pipeline) ProcessReports(ctx context.Context, w http.ResponseWriter, r *http.Request) *ReportBatch {
+// report. Returns true if the request was dropped due to a full queue and false
```

Comment:
This return semantics feel backwards. I'd expect this to return true if things succeed.

Also, since there seem to be multiple ways this can fail (not just if the queue is full) you might make this return an error instead.

Reply:
My concern is isolating drops due to a full queue as compared to every other type of error (e.g. bad request, not a POST, etc.). The error syntax is captured in the http.Error calls. The bool returned is simply whether or not the queue was full. I'm open to a better way to capture this.

Reply:
I fixed this to return an error, and added a special error for when it is dropped so that the caller can determine if it was dropped, successful, or something else.